syntex/src/lib.rs

```diff
@@ -150,12 +149,50 @@ impl Registry {
 
         let sess = parse::ParseSess::new();
 
-        let mut krate = parse::parse_crate_from_file(
+        let krate = parse::parse_crate_from_file(
             src,
-            self.cfg,
+            self.cfg.clone(),
             &sess).unwrap();
 
-        krate.attrs.extend(self.attrs);
+        let src_name = src.to_str().unwrap().to_string();
+
+        let out = try!(self.expand_crate(
+                crate_name.to_string(),
+                &sess,
+                src_name,
+                krate));
+
+        let mut dst = try!(File::create(dst));
+        dst.write_all(&out)
+    }
+
+    /// This method will expand all macros in the source string `src`, and return the results in a
+    /// string.
+    pub fn expand_str(self,
```

Comment:
The other function uses `crate_name: &str`. Could the three arguments all be `&str` instead of `String`?